Spigot-Server-Patches/0524-Added-firing-of-PlayerChangeBeaconEffectEvent.patch

```diff
@@ -0,0 +1,57 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Wed, 24 Jun 2020 15:14:51 -0600
+Subject: [PATCH] Added firing of PlayerChangeBeaconEffectEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
+index 10e3bf60f13fbc8e26071bcfe8d19cbb635662a9..77c23c2f38c0778fe01820366612fc9b17414f35 100644
+--- a/src/main/java/net/minecraft/server/PlayerConnection.java
++++ b/src/main/java/net/minecraft/server/PlayerConnection.java
+@@ -1,5 +1,6 @@
+ package net.minecraft.server;
+ 
++import com.destroystokyo.paper.event.player.PlayerChangeBeaconEffectEvent;
+ import com.google.common.primitives.Doubles;
+ import com.google.common.primitives.Floats;
+ import com.mojang.brigadier.ParseResults;
+@@ -26,6 +27,7 @@ import org.bukkit.Location;
+ import org.bukkit.craftbukkit.entity.CraftPlayer;
+ import org.bukkit.craftbukkit.event.CraftEventFactory;
+ import org.bukkit.craftbukkit.inventory.CraftItemStack;
++import org.bukkit.craftbukkit.potion.CraftPotionEffectType;
+ import org.bukkit.craftbukkit.util.CraftChatMessage;
+ import org.bukkit.craftbukkit.util.CraftMagicNumbers;
+ import org.bukkit.craftbukkit.util.LazyPlayerSet;
+@@ -58,6 +60,8 @@ import org.bukkit.event.player.PlayerToggleSprintEvent;
+ import org.bukkit.inventory.CraftingInventory;
+ import org.bukkit.inventory.EquipmentSlot;
+ import org.bukkit.inventory.InventoryView;
++import org.bukkit.potion.PotionEffect;
++import org.bukkit.potion.PotionEffectType;
+ import org.bukkit.util.NumberConversions;
+ import com.destroystokyo.paper.event.player.IllegalPacketEvent; // Paper
+ import com.destroystokyo.paper.event.player.PlayerJumpEvent; // Paper
+@@ -728,9 +732,20 @@ public class PlayerConnection implements PacketListenerPlayIn {
+     public void a(PacketPlayInBeacon packetplayinbeacon) {
+         PlayerConnectionUtils.ensureMainThread(packetplayinbeacon, this, this.player.getWorldServer());
+         if (this.player.activeContainer instanceof ContainerBeacon) {
+-            ((ContainerBeacon) this.player.activeContainer).c(packetplayinbeacon.b(), packetplayinbeacon.c());
+-        }
++            ContainerBeacon containerBeacon = (ContainerBeacon) this.player.activeContainer;
```

Comment:
Unless I'm missing something, this should be commented with Paper start & Paper end.